Sources/TuistGenerator/Generator/BuildPhaseGenerator.swift

```diff
@@ -217,6 +218,7 @@ final class BuildPhaseGenerator: BuildPhaseGenerating {
     func generateSourcesBuildPhase(
         files: [SourceFile],
         coreDataModels: [CoreDataModel],
+        target: Target,
```

Comment:
We may not need to address this with this work, but seeing this needing to be passed in is making me think that we may be able to move the `applicableTo` logic into the mapping phase now.    For this PR i think this is fine but something we can think on for future work.